cmd/gitserver/internal/perforce/protects.go

```
@@ -151,6 +186,12 @@ func parseP4Protects(out []byte) ([]*p4types.Protect, error) {
 			return nil, errors.Wrap(err, "failed to unmarshal protect line")
 		}
 
+		if parsedLine.DepotFile == "" {
```

I wonder if we should turn this on with an env var - as it requires that script to be correctly deployed anyways.

In addition, is there something we could do here to make sure things that don't parse properly (i.e. this json thing that doesn't contain any data) will not cause all perms to be dropped, and fail instead?

---

We could always simply return an error if DepotFile is empty. I don't know what we would do with an empty DepotFile field anyways?

---

Yeah I guess that’s good, better than this silent failure we had here